win32/src/win32evtlog.i

```diff
@@ -511,6 +512,113 @@ static PyObject *PyEvtOpenChannelEnum(PyObject *self, PyObject *args, PyObject *
 }
 PyCFunction pfnPyEvtOpenChannelEnum = (PyCFunction) PyEvtOpenChannelEnum;
 
+
+//Helper function to convert a list of strings double zero terminated
+//into a Python list of strings.
+//e.g. hello\0world\0\0 -> [ "hello", "world" ]
+static PyObject* PyList_FromDoubleTerminatedWSTR( LPWSTR strings )
+{
+	PyObject* ret = PyList_New( 0 );
+	WCHAR* cur = strings;
+	while ( *cur ) {
+		PyObject* keyword = PyWinObject_FromWCHAR( cur );
+		PyList_Append( ret, keyword );
+		Py_DECREF( keyword );
+		cur += wcslen( cur ) + 1;
+	}
+
+	return ret;
```

Comment:
This function should handle failure a bit better - `PyList_New` and `PyWinObject_FromWCHAR` might return NULL - it's probably enough to early return if `ret` is null and to use `Py_XDECREF()` (although I guess even the former isn't strictly necessary as `PyList_Append` will not crash if either arg is null - but Py_DECREF will)